clang/include/clang/Basic/TargetInfo.h

```diff
@@ -1398,6 +1400,42 @@ class TargetInfo : public TransferrableTargetInfo,
       }
       llvm_unreachable("Unexpected SignReturnAddressKeyKind");
     }
+
+  public:
+    BranchProtectionInfo() = default;
```

Comment:
Do we still need the default constructor, or no?

Reply:
We need the default constructor:
e.g https://github.com/llvm/llvm-project/blob/main/clang/lib/Sema/SemaDeclAttr.cpp#L3481 
https://github.com/llvm/llvm-project/blob/main/clang/lib/CodeGen/Targets/ARM.cpp#L144